x/crosschain/keeper/msg_add_to_intx_tracker.go

```diff
@@ -0,0 +1,116 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	eth "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+	observerTypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+func (k msgServer) AddToInTxTracker(goCtx context.Context, msg *types.MsgAddToInTxTracker) (*types.MsgAddToInTxTrackerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	chain := k.zetaObserverKeeper.GetParams(ctx).GetChainFromChainID(msg.ChainId)
+	if chain == nil {
+		return nil, observerTypes.ErrSupportedChains
+	}
+
+	adminPolicyAccount := k.zetaObserverKeeper.GetParams(ctx).GetAdminPolicyAccount(observerTypes.Policy_Type_group1)
+	isAdmin := msg.Creator == adminPolicyAccount
+
+	isObserver, err := k.zetaObserverKeeper.IsAuthorized(ctx, msg.Creator, chain)
+	if err != nil {
+		ctx.Logger().Error("Error while checking if the account is an observer", err)
+	}
+	isProven := false
+	if !(isAdmin || isObserver) && msg.Proof != nil {
+		isProven, err = k.VerifyInTxTrackerProof(ctx, msg.Proof, msg.BlockHash, msg.TxIndex, msg.ChainId, msg.CoinType)
```

Comment:
Following the discussion we had, we want to be able to disable it. Maybe it can be done in this PR?
I think the best would be to add it as a crosschain flag
https://github.com/zeta-chain/node/blob/develop/proto/observer/crosschain_flags.proto

Ideally, the check could be in a unique place common for all components that needs to verify a header, but I think it requires  https://github.com/zeta-chain/node/issues/1254 to be completed